lib/courier/stores/agent.ex

```diff
@@ -0,0 +1,60 @@
+defmodule Courier.Stores.Agent do
+  defmacro __using__(_) do
+    quote do
+      def start_link() do
+        Agent.start_link(fn -> [] end, name: __MODULE__)
+      end
+
+      def all(past: true) do
+        Agent.get(__MODULE__, &before_now/1)
+      end
+
+      def all() do
+        Agent.get(__MODULE__, &strip_timestamps/1)
+      end
+
+      def put({%Mail.Message{} = message, timestamp}) do
+        Agent.update(__MODULE__, fn(messages) -> [{message, timestamp} | messages] end)
+      end
+
+      def delete(%Mail.Message{} = message) do
+        Agent.update(__MODULE__, fn(messages) -> 
+          messages
+          |> Enum.find_index(&(message == elem(&1, 0)))
```

Comment:
A second thought would be to use a `MapSet` instead of a list if we don't have a unique key. MapSet's give us everything we're using here with a list but constant access and deletion.